test/blackbox-tests/test-cases/directory-targets/cache-file-and-dir.t

```diff
@@ -0,0 +1,56 @@
+This checks what happens when a file available in the cache is used in a directory target.
+
+  $ export DUNE_CACHE_ROOT=$PWD/.cache
+  $ export DUNE_CACHE=enabled
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.11)
+  > (using directory-targets 0.1)
+  > EOF
+
+  $ cat > dune << EOF
+  > (rule
+  >  (with-stdout-to file_out (run ./gen.exe)))
+  > 
+  > (rule
+  >  (target (dir dir_out))
+  >  (deps gen.exe)
+  >  (action
+  >   (no-infer
+  >    (progn
+  >     (run mkdir dir_out)
+  >     (with-stdout-to dir_out/a (run ./gen.exe))
+  >     (write-file dir_out/b contents_b)))))
+  > 
+  > (executable
```

Comment:
Instead of building an executable, you could:

1. execute it with `ocaml gen.ml`
2. use a shell script (we already depend on a shell script elsewhere in this test anyway)

I feel it's a bit necessary to make use of dune's OCaml rules for this test.